lib/chef/resource/windows_pagefile.rb

```diff
@@ -82,14 +91,14 @@ class WindowsPagefile < Chef::Resource
 
         if automatic_managed
           set_automatic_managed unless automatic_managed?
-        else
+        elsif automatic_managed == false
           unset_automatic_managed if automatic_managed?
+        else
+          validate_name
+          pagefile_drive_exist?(pagefile)
```

Comment:
this returns a boolean but it doesn't look like you are doing anything with it

Reply:
oh I see it raises. Is that idiomatic ruby? Would a `assert_pagefile_drive_exist!` be more appropriate?

Reply:
Yeah, I need to tweak that. What I was trying to capture was having the method raise an exception if you tried to create a pagefile on H: drive if the physical drive didn't exist. Stupid PowerShell will happily let you create a pagefile named h:\pageile.sys even if you don't actually have a H drive.